src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLogger.cs

```diff
@@ -32,7 +34,7 @@ internal class ApplicationInsightsLogger : ILogger
         internal const string MetricMaxKey = "max";
         internal const string MetricStandardDeviationKey = "standarddeviation";
 
-        private static readonly string[] SystemScopeKeys =
+        private static readonly HashSet<string> SystemScopeKeys = new HashSet<string>
```

Comment:
We do lookups for every scope property on each log. E.g. for a single function execution we end up comparing about 25 scope properties, and each was a lookup in this long string array.